docs/6-the-headliner/3-advanced-deployments.md

```diff
@@ -1,29 +1,98 @@
+
+    <div class="highlight" style="background: #f7f7f7">
+    <pre><code class="language-yaml">
+      ---
+      apiVersion: serving.kserve.io/v1beta1
+      kind: InferenceService
+      metadata:
+        name: jukebox
+        annotations:
+          openshift.io/display-name: jukebox
+          serving.knative.openshift.io/enablePassthrough: 'true'
+          sidecar.istio.io/inject: 'true'
+          sidecar.istio.io/rewriteAppHTTPProbers: 'true'
+          autoscaling.knative.dev/target: "1"
+        finalizers:
+          - inferenceservice.finalizers
+        labels:
+          opendatahub.io/dashboard: 'true'
+      spec:
+        predictor:
+          canaryTrafficPercent: 20 ### 👈 this does the magic 🔮
+       ...
+    </code></pre></div>
+
+
+2. Let's push the change. This will redeploy the previous version automatically do split the traffic based on the percentage we provided above. 
 
     ```bash
-    oc get isvc -n <USER_NAME>-test
+    cd /opt/app-root/src/mlops-gitops
+    git pull
+    git add .
+    git commit -m  "🦜 UPDATE - canary deployment enabled 🦜"
+    git push
     ```
 
+3. After Argo CD syncs the changes, check the output of previous command again.
+   
     ```bash
-        NAME      URL                                                 READY   PREV   LATEST   PREVROLLEDOUTREVISION     LATESTREADYREVISION       AGE
-        jukebox   https://jukebox-<USER_NAME>-test.<CLUSTER_DOMAIN>   True    90     10       jukebox-predictor-00013   jukebox-predictor-00014   25h
-    ``` 
+    oc get isvc -n <USER_NAME>-test
+    ```
+
+    You should get an output like this:
+
+    <div class="highlight" style="background: #f7f7f7; overflow-x: auto; padding: 10px;">
+    <pre><code class="language-bash">
+        NAME      URL                                                                           READY   PREV   LATEST   PREVROLLEDOUTREVISION     LATESTREADYREVISION       AGE
+        jukebox   https://jukebox-<USER_NAME>-test.<CLUSTER_DOMAIN>   True    80     20       jukebox-predictor-00002   jukebox-predictor-00003   25h
+   </code></pre></div>
 
 
-    Let's use the locust script again to generate some traffic and verify the both deployments receive traffic. We can use on monitoring for that.
+4. Let's check if we are really able to send 20% of the traffic to the latest version while the rest of the traffic is being handled by the previous version. We can rely on `locust` to generate some traffic again. Let's go back to Jupyter Notebook and open `jukebox/6-advanced_deployments/2-canary_testing.ipynb`
 
-    We can incrementally increase the traffic percantage towards the latest version by updating the values file and set it `100` when we decide to use the latest version only. 
+5. After you start generating the traffic, let's go to [Grafana](https://jukebox-grafana-route-<USER_NAME>-mlops.<CLUSTER_DOMAIN>/) you previously deployed.
 
+  ```bash
+  https://jukebox-grafana-route-<USER_NAME>-mlops.<CLUSTER_DOMAIN>/
+  ```
```

Comment:
Should we add a section to change to 100% for the new revision "after we have seen it working" so the 80/20 split doesn't affect the next section?

Reply:
haha yeah that's the plan, and then take 100% all back to the previous revision to explain Blue / Green approach (or to show that KServe holds the revisions)..however I just want this to merged and will send an update on it later :D